src/interface/extension_interface.hpp

```diff
@@ -240,6 +335,34 @@ typename sb_handle_t::event_t _matcopy(sb_handle_t& sb_handle, char trans,
   }
 }
 
+template <typename sb_handle_t, typename element_t, typename index_t,
+          typename container_t>
+typename sb_handle_t::event_t _omatadd(sb_handle_t& sb_handle, char trans_a,
+                                       char trans_b, index_t m, index_t n,
+                                       element_t alpha, container_t a,
+                                       index_t lda, element_t beta,
+                                       container_t b, index_t ldb,
+                                       container_t c, index_t ldc) {
+  if (trans_a == 't') {
+    if (trans_b == 't') {
+      return _omatadd_impl<true, true>(sb_handle, m, n, alpha, a, lda, beta, b,
+                                       ldb, c, ldc);
+    } else {
+      return _omatadd_impl<true, false>(sb_handle, m, n, alpha, a, lda, beta, b,
+                                        ldb, c, ldc);
+    }
+  } else if (trans_b == 't') {
+    // In this case, (alpha,a) & (beta,b) parameters positions are swapped as
+    // the kernel implementation assumes the first input matrix is the
+    // transposed one for simplicity purposes.
```

Comment:
Could you briefly explain why in case one of the operand is transposed it needs to be passed as the first one?

Reply:
Yes, I suggest adding a couple of lines for explaining it in `transpose.hpp` and maybe reference here to that explanation.